### PR TITLE
'Gulp docs', deprecated server arguments, doc updates

### PR DESCRIPTION
**Pull request by scottdixon-zz**
Continuing https://github.com/appium/appium/pull/5900
- Remove grunt.md, replaced with gulp.md
- Translated `grunt docs` to `gulp docs`
- Deprecated server arguments

I'll reach out to lihuazhang once this is merged to see if they're available to update the cn version.

@imurchie

### CODE REVIEW

File: docs/en/contributing-to-appium/gulp.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+[Gulp](http://gulpjs.com) is like make for Node.js! We use it to automate all
+kinds of appium dev tasks. Here's what you can do:
+
+|Task|Description|
+|----|-----------|
+|gulp once|Cleans, lints, transpiles and runs unit tests|
+|gulp watch|Automatically runs `gulp once` when code changes|
+|gulp lint|Runs JSLint|
+|gulp jshint|Runs JSHint|
+|gulp transpile|Transpiles our ES7 code to ES5, generates `/build` directory + contents|
```
Comment (imurchie) [inline docs/en/contributing-to-appium/gulp.md]: ES7/ES2015

File: docs/en/writing-running-appium/server-args.md [hunk 2]
```diff
@@ -12,78 +12,79 @@ All flags are optional, but some are required in conjunction with certain others
...
-|`--intent-flags`|0x10200000|(Android-only) Flags that will be used to start activity|`--intent-flags 0x10200000`|
-|`--intent-args`|null|(Android-only) Additional intent arguments that will be used to start activity|`--intent-args 0x10200000`|
-|`--dont-stop-app-on-reset`|false|(Android-only) When included, refrains from stopping the app before restart||
 |`--debug-log-spacing`|false|Add exaggerated spacing in logs to help with visual inspection||
 |`--suppress-adb-kill-server`|false|(Android-only) If set, prevents Appium from killing the adb server instance||
-|`--webkit-debug-proxy-port`|false|(IOS-only) Local port used for communication with ios-webkit-debug-proxy|`--webkit-debug-proxy-port 27753`||
+|`--async-trace`|false|Add long stack traces to log entries. Recommended for debugging only.||
+|`--webkit-debug-proxy-port`|27753|(IOS-only) Local port used for communication with ios-webkit-debug-proxy|`--webkit-debug-proxy-port 27753`|
+|`--default-capabilities`|[object Object]|Set the default desired capabilities, which will be set on each session unless overridden by recieved capabilities.|`--default-capabilities {"app": "myapp.app", "deviceName": "iPhone Simulator"}`|
```
Comment (imurchie) [inline docs/en/writing-running-appium/server-args.md]: Something is wrong in generating this, leading to `[object Object]`.

**Comment by imurchie** [comment]
We still need some description of how to use `--default-capabilities`, which now takes the place of all the old server flags that are also desired capabilities. It takes a JSON string.

**Comment by scottdixon** [comment]
In addition to line 53 @imurchie ?

**Comment by imurchie** [comment]
I think, yes. It is currently unclear that all the newly-deprecated arguments can still be specified by putting them in the `--default-capabilities` object.

File: docs/en/contributing-to-appium/gulp.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+|gulp unit-test|Runs unit tests|
+|gulp e2e-test|Runs e2e tests|
+|gulp docs|Generates docs/en/writing-running-appium/server-args.md doc|
```
Comment (jlipps) [inline docs/en/contributing-to-appium/gulp.md]: does this actually still work I wonder?

File: docs/en/writing-running-appium/server-args.md [hunk 4]
```diff
@@ -12,78 +12,79 @@ All flags are optional, but some are required in conjunction with certain others
...
+|`--app-wait-activity`|false|[DEPRECATED] - (Android-only) Activity name for the Android activity you want to wait for (e.g., SplashActivity)|`--app-wait-activity SplashActivity`|
+|`--device-ready-timeout`|5|[DEPRECATED] - (Android-only) Timeout in seconds while waiting for device to become ready|`--device-ready-timeout 5`|
+|`--android-coverage`|false|[DEPRECATED] - (Android-only) Fully qualified instrumentation class. Passed to -w in adb shell am instrument -e coverage true -w |`--android-coverage com.my.Pkg/com.my.Pkg.instrumentation.MyInstrumentation`|
+|`--avd`|null|[DEPRECATED] - (Android-only) Name of the avd to launch|`--avd @default`|
+|`--avd-args`|null|[DEPRECATED] - (Android-only) Additional emulator arguments to launch the avd|`--avd-args -no-snapshot-load`|
+|`--use-keystore`|false|[DEPRECATED] - (Android-only) When set the keystore will be used to sign apks.||
+|`--keystore-path`|/Users/scottdixon/.android/debug.keystore|[DEPRECATED] - (Android-only) Path to keystore||
```
Comment (jlipps) [inline docs/en/writing-running-appium/server-args.md]: unfortunately the auto generation will put your username here. we might want to replace it with `<user>` or something

File: gulpfile.js [hunk 5]
```diff
@@ -10,3 +12,35 @@ boilerplate({
     files: ['${testDir}/**/*-specs.js']
   },
 });
+
+// generates server arguments readme
+gulp.task('docs', ['transpile'], function () {
```
Comment (jlipps) [inline gulpfile.js]: oh i see you ported it, excellent :-)

**Comment by jlipps** [comment]
couple small comments, otherwise looks good! did you read through the docs as a whole to make sure things still generally make sense?

**Comment by scottdixon** [comment]
@imurchie - updated some docs, please run your eyes over them.

**Comment by imurchie** [comment]
LGTM.

**Comment by scottdixon** [comment]
yep @jlipps, combined with https://github.com/appium/appium/pull/5900 it still makes sense for the most part - main changes: Xcode/iOS versions, grunt/gulp change. Is ES2015 style guide still a todo?

**Comment by jlipps** [comment]
yes however we can leave that for later

File: docs/en/writing-running-appium/server-args.md [hunk 6]
```diff
@@ -12,78 +12,79 @@ All flags are optional, but some are required in conjunction with certain others
...
-|`--intent-flags`|0x10200000|(Android-only) Flags that will be used to start activity|`--intent-flags 0x10200000`|
-|`--intent-args`|null|(Android-only) Additional intent arguments that will be used to start activity|`--intent-args 0x10200000`|
-|`--dont-stop-app-on-reset`|false|(Android-only) When included, refrains from stopping the app before restart||
 |`--debug-log-spacing`|false|Add exaggerated spacing in logs to help with visual inspection||
 |`--suppress-adb-kill-server`|false|(Android-only) If set, prevents Appium from killing the adb server instance||
-|`--webkit-debug-proxy-port`|false|(IOS-only) Local port used for communication with ios-webkit-debug-proxy|`--webkit-debug-proxy-port 27753`||
+|`--async-trace`|false|Add long stack traces to log entries. Recommended for debugging only.||
+|`--webkit-debug-proxy-port`|27753|(IOS-only) Local port used for communication with ios-webkit-debug-proxy|`--webkit-debug-proxy-port 27753`|
+|`--default-capabilities`|{}|Set the default desired capabilities, which will be set on each session unless overridden by recieved capabilities.|`--default-capabilities {"app": "myapp.app", "deviceName": "iPhone Simulator"}`|
```
Comment (jlipps) [inline docs/en/writing-running-appium/server-args.md]: slight typo here, recieved -> received

**Pull request merged**